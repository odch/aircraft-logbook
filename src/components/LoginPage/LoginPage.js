import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import withStyles from '@material-ui/core/styles/withStyles'
import LoginForm from '../../containers/LoginFormContainer'

const styles = theme => ({
  layout: {
    width: 'auto',
    display: 'block',
    marginTop: theme.spacing.unit * 8,
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  registrationButton: {
    marginTop: '1em'
  }
})

class LoginPage extends React.Component {
  render() {
    if (!this.props.auth.isEmpty) {
      return <Redirect to="/" />
    }

    const { classes } = this.props

    return (
      <main className={classes.layout}>
        <LoginForm />
        <Button
          className={classes.registrationButton}
          href="/register"
          variant="raised"
          color="primary"
          data-cy="registration"
          fullWidth
        >
          <FormattedMessage id="login.registration" />
        </Button>
      </main>
    )
  }
}

LoginPage.propTypes = {
  auth: PropTypes.shape({
    isEmpty: PropTypes.bool.isRequired
  }),
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(LoginPage)
