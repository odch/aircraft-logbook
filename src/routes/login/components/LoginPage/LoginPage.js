import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import featureToggles from 'feature-toggles'
import GoogleLogin from '../../containers/GoogleLoginContainer'
import LoginForm from '../../containers/LoginFormContainer'
import getAuthQueryToken from '../../../../util/getAuthQueryToken'
import LoadingIcon from '../../../../components/LoadingIcon'

const styles = theme => ({
  layout: {
    width: 'auto',
    display: 'block',
    marginTop: theme.spacing(8),
    marginLeft: theme.spacing(3),
    marginRight: theme.spacing(3),
    [theme.breakpoints.up(400 + theme.spacing(3 * 2))]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  registrationButton: {
    marginTop: '1em'
  },
  google: {
    marginBottom: '2em'
  }
})

class LoginPage extends React.Component {
  componentDidMount() {
    const { auth, router, loginWithToken } = this.props
    if (auth.isEmpty) {
      const queryToken = getAuthQueryToken(router.location)
      if (queryToken) {
        loginWithToken(queryToken)
      }
    }
  }

  render() {
    const { auth, router, tokenLogin, classes } = this.props

    if (!auth.isEmpty) {
      if (router.location.state && router.location.state.from) {
        return <Redirect to={router.location.state.from.pathname} />
      }
      return <Redirect to="/" />
    }

    if (tokenLogin.submitted) {
      return <LoadingIcon />
    }

    if (tokenLogin.failed) {
      return (
        <main className={classes.layout}>
          <FormattedMessage id="login.tokeninvalid" />
        </main>
      )
    }

    return (
      <main className={classes.layout}>
        <GoogleLogin className={classes.google} />
        {featureToggles.isFeatureEnabled('emailPasswordAuth') && (
          <React.Fragment>
            <Typography align="center" color="textSecondary" gutterBottom>
              <FormattedMessage id="login.or" />
            </Typography>
            <LoginForm />
            {featureToggles.isFeatureEnabled('registration') && (
              <Button
                className={classes.registrationButton}
                href="/register"
                variant="text"
                color="primary"
                data-cy="registration"
                fullWidth
              >
                <FormattedMessage id="login.registration" />
              </Button>
            )}
          </React.Fragment>
        )}
      </main>
    )
  }
}

LoginPage.propTypes = {
  auth: PropTypes.shape({
    isEmpty: PropTypes.bool.isRequired
  }).isRequired,
  classes: PropTypes.object.isRequired,
  router: PropTypes.shape({
    location: PropTypes.shape({
      search: PropTypes.string,
      state: PropTypes.shape({
        queryToken: PropTypes.shape({
          orgId: PropTypes.string.isRequired,
          token: PropTypes.string.isRequired
        }),
        from: PropTypes.shape({
          pathname: PropTypes.string.isRequired
        })
      })
    }).isRequired
  }).isRequired,
  tokenLogin: PropTypes.shape({
    submitted: PropTypes.bool.isRequired,
    failed: PropTypes.bool.isRequired
  }).isRequired,
  loginWithToken: PropTypes.func.isRequired
}

export default withStyles(styles)(LoginPage)
