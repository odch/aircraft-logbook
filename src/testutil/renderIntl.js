import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
