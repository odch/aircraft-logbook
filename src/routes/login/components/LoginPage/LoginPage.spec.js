import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import renderer from 'react-test-renderer'
import renderIntl from '../../../../testutil/renderIntl'
import configureStore from 'redux-mock-store'
import LoginPage from './LoginPage'

const StartPage = () => <div>start page</div>

describe('components', () => {
  describe('LoginPage', () => {
    it('redirects to start page if authenticated', () => {
      const auth = {
        isEmpty: false
      }
      const store = configureStore()({
        firebase: {
          auth
        }
      })
      const tree = renderer
        .create(
          <Provider store={store}>
            <Router>
              <Switch>
                <Route exact path="/" component={StartPage} />
                <LoginPage auth={auth} />
              </Switch>
            </Router>
          </Provider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders login form if not authenticated', () => {
      const auth = {
        isEmpty: true
      }
      const store = configureStore()({
        firebase: {
          auth
        },
        login: {
          username: 'test@example.com',
          password: 'mypassword',
          failed: false,
          submitted: false
        }
      })
      const tree = renderIntl(
        <Provider store={store}>
          <Router>
            <Switch>
              <LoginPage auth={auth} />
              <Route exact path="/" component={StartPage} />
            </Switch>
          </Router>
        </Provider>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
