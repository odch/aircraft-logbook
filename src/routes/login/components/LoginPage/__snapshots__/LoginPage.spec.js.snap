// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LoginPage redirects to from state if authenticated 1`] = `
<div>
  org detail page
</div>
`;

exports[`components LoginPage redirects to start page as default if authenticated 1`] = `
<div>
  start page
</div>
`;

exports[`components LoginPage renders login form if not authenticated 1`] = `
<main
  className="LoginPage-layout-7"
>
  <div
    className="LoginPage-google-9"
  >
    <button
      className="MuiButtonBase-root MuiButton-root GoogleLogin-button-11 MuiButton-contained MuiButton-sizeLarge MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <img
          alt="Google Logo"
          className="GoogleLogin-logo-12"
          src="test-file-stub"
        />
        Mit Google anmelden
      </span>
    </button>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    oder
  </p>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="email"
      >
        E-Mail
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-cy="email"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={true}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value="test@example.com"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="password"
      >
        Passwort
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-cy="password"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value="mypassword"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      data-cy="submit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Anmelden
      </span>
    </button>
  </form>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiButton-root LoginPage-registrationButton-8 MuiButton-text MuiButton-textPrimary MuiButton-fullWidth"
    data-cy="registration"
    href="/register"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiButton-label"
    >
      Noch kein Konto? Hier registrieren!
    </span>
  </a>
</main>
`;
